### PR TITLE
Added '@ref' to Camera Calibration and 3D Reconstruction documentation

### PR DESCRIPTION
**Pull request by IanMaquignaz**
Added '@ref' to enumerations to enable doxygen checks in Camera Calibration and 3D Reconstruction documentation.

<cut/>

**Notes**:
* Did not check enumerations with '#'. 
* Could not add `@ref` to enumerations in `@code` blocks.

**Observations** (not addressed)
* stereoCalibrate(...) and fisheye::stereoCalibrate(...) has random '?' in description? [See here](https://docs.opencv.org/3.4/d9/d0c/group__calib3d.html#ga91018d80e2a93ade37539f01e6f07de5) and [here](https://docs.opencv.org/3.4/db/d58/group__calib3d__fisheye.html#gadbb3a6ca6429528ef302c784df47949b)
_CALIB_FIX_INTRINSIC Fix cameraMatrix? and distCoeffs? so that only R, T, E, and F matrices are estimated._
* fisheye::stereoCalibrate() has no enumeration option CALIB_ZERO_DISPARITY. [See here](https://docs.opencv.org/3.4/db/d58/group__calib3d__fisheye.html#gac1af58774006689056b0f2ef1db55ecc)

### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or other license that is incompatible with OpenCV
- [x] The PR is proposed to proper branch
- [x] There is reference to original bug report and related work
- [x] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [x] The feature is well documented and sample code can be built with the project CMake

```
force_builders_only=docs,linux
```

### CODE REVIEW

**Comment by IanMaquignaz** [comment]
Not sure what the failed build is all about... I only modified the Docs which built with [the following warnings:](https://pullrequest.opencv.org/buildbot/builders/precommit_docs/builds/27844/steps/make%20doxygen/logs/warnings%20%2822%29)

Couple of the warnings stem from half of this enum disapearing:
`enum { CALIB_CB_ADAPTIVE_THRESH = 1,
       CALIB_CB_NORMALIZE_IMAGE = 2,
       CALIB_CB_FILTER_QUADS    = 4,
       CALIB_CB_FAST_CHECK      = 8,
       CALIB_CB_EXHAUSTIVE      = 16,
       CALIB_CB_ACCURACY        = 32,
       CALIB_CB_LARGER          = 64,
       CALIB_CB_MARKER          = 128
     };`
I assume running `git rebase -i --onto upstream/3.4 upstream/master` produced this result?

**Comment by alalek** [comment]
> what the failed

Save current work and perform step-by-step small updates. There are too many changes on Diff page.

File: modules/calib3d/include/opencv2/calib3d.hpp [hunk 1]
```diff
@@ -593,19 +593,19 @@ pairs to compute an initial homography estimate with a simple least-squares sche
 
 However, if not all of the point pairs ( \f$srcPoints_i\f$, \f$dstPoints_i\f$ ) fit the rigid perspective
 transformation (that is, there are some outliers), this initial estimate will be poor. In this case,
-you can use one of the three robust methods. The methods RANSAC, LMeDS and RHO try many different
+you can use one of the three robust methods. The methods RANSAC, LMEDS and RHO try many different
```
Comment (catree) [inline modules/calib3d/include/opencv2/calib3d.hpp:596]: In my opinion, when referring to the method we should keep `LMeDS`/`LMedS`.

See [Least Median of Squares](https://www-sop.inria.fr/odyssee/software/old_robotvis/Tutorial-Estim/node25.html#SECTION000105000000000000000) from a tutorial by Zhengyou Zhang on [Parameter Estimation Techniques: A Tutorial with Application to Conic Fitting](https://www-sop.inria.fr/odyssee/software/old_robotvis/Tutorial-Estim/Main.html), published in October 1995.

**Comment by IanMaquignaz** [comment]
OK, so... code has been rebuilt locally a few times now and code builds fine with OpenCV master. The build issues seen here stem from the rebase (`git rebase -i --onto upstream/3.4 upstream/master`), where enums and parameters have changed to reflect what's shown below in 3.4. 

[In Master](https://github.com/opencv/opencv/blob/master/modules/calib3d/include/opencv2/calib3d.hpp):
`//! type of the robust estimation algorithm
enum { LMEDS  = 4,  //!< least-median of squares algorithm
       RANSAC = 8,  //!< RANSAC algorithm
       RHO    = 16, //!< RHO algorithm
       USAC_DEFAULT  = 32, //!< USAC algorithm, default settings
       USAC_PARALLEL = 33, //!< USAC, parallel version
       USAC_FM_8PTS = 34,  //!< USAC, fundamental matrix 8 points
       USAC_FAST = 35,     //!< USAC, fast settings
       USAC_ACCURATE = 36, //!< USAC, accurate settings
       USAC_PROSAC = 37,   //!< USAC, sorted points, runs PROSAC
       USAC_MAGSAC = 38    //!< USAC, runs MAGSAC++
     };
`

[In 3.4](https://github.com/opencv/opencv/blob/3.4/modules/calib3d/include/opencv2/calib3d.hpp):
`//! type of the robust estimation algorithm
enum { LMEDS  = 4, //!< least-median of squares algorithm
       RANSAC = 8, //!< RANSAC algorithm
       RHO    = 16 //!< RHO algorithm
     };
`

The question therefore is should I be rebasing to merge with 3.4? and if yes, how do I approach this? I've identified one USAC function declaration which appears without reference (see below), but thus far my attempts at a build without this header has only produced further errors downstream. 

` CV_EXPORTS_W Mat findEssentialMat( InputArray points1, InputArray points2,
                       InputArray cameraMatrix1, InputArray cameraMatrix2,
                       InputArray dist_coeff1, InputArray dist_coeff2, OutputArray mask,
                       const UsacParams &params);`

**Comment by alalek** [comment]
Need to create 2 PRs:
- keep this PR for 3.4: common changes and specific for 3.4 version
- separate PR for master: specific changes for master version (not necessary to duplicate "common" changes from 3.4 - they will be automatically merged during regular "Merge 3.4" process)

**Comment by IanMaquignaz** [comment]
> Need to create 2 PRs

If there's no way to merge my current working alterations in master into 3.4 without painstakingly re-editing calid3d.hpp's 3467 lines then I'm not sure I have the time to complete this task. 

Alternatively, I've pushed my latest rebase revision which compiles locally for 3.4. Though functional, it reintroduces documentation for `findChessboardCornersSB()` which is currently omitted from 3.4. Is this latest commit something which would be acceptable? It can be tweaked, but if everything can be done in 3.4 then that would be ideal. No need to identify un-applied changes to redo in master.

**Comment by IanMaquignaz** [comment]
@alalek how do you get the opencv-pushbot to rebuilt everything not just the docs?

**Comment by alalek** [comment]
> findChessboardCornersSB()

should be removed from this patch (because it goes to 3.4 branch).
Look on "Files changed" tab on PR page and check for unrelated/unwanted changes.

---

> rebuilt everything

I reduced testing scope (PR is under active development, so full scope is not really necessary).
Let fix docs and linux builders first.

**Comment by asmorkalov** [comment]
@IanMaquignaz Do you have any progress with the patch?

**Comment by IanMaquignaz** [comment]
@asmorkalov I have not had a chance to work on this since my last push. That being said, to accomplish the changes suggested by alalek, I need to create a fresh branch and restart. 

I may have some time next week.

**Comment by IanMaquignaz** [comment]
See new PR #19089

**Comment by asmorkalov** [comment]
The patch is replaced with more focused PRs for 3.4 and master .

**Pull request closed without merging**